github.com.hashicorp.go.kms.wrapping.v2.types.proto

```diff
@@ -0,0 +1,95 @@
+syntax = "proto3";
+
+option go_package = "github.com/hashicorp/go-kms-wrapping/v2;wrapping";
+
+package github.com.hashicorp.go.kms.wrapping.v2.types;
+
+import "google/protobuf/struct.proto";
+
+// WrapperConfig is the result of a call to SetConfig on a wrapper, returning
+// relevant information about the wrapper and its updated configuration
+message WrapperConfig {
+	map<string, string> metadata = 10;
+}
+
+// EnvelopeInfo contains the information necessary to perfom encryption or
+// decryption in an envelope fashion
+message EnvelopeInfo {
+	// Ciphertext is the ciphertext from the envelope
+	bytes ciphertext = 1;
+
+	// Key is the key used in the envelope
+	bytes key = 2;
+
+	// IV is the initialization value used during encryption in the envelope
+	bytes iv = 3;
+}
+
+// BlobInfo contains information about the encrypted value along with
+// information about the key used to encrypt it
+message BlobInfo {
+	// Ciphertext is the encrypted bytes
+	bytes ciphertext = 1;
+
+	// IV is the initialization value used during encryption
+	bytes iv = 2;
+
+	// HMAC is the bytes of the HMAC, if any
+	bytes hmac = 3;
+
+	// Wrapped can be used by the client to indicate whether Ciphertext actually
+	// contains wrapped data or not. This can be useful if you want to reuse the
+	// same struct to pass data along before and after wrapping. Deprecated in
+	// favor of plaintext.
+	bool wrapped = 4 [deprecated = true];
```

Comment:
This will be a go.mod v2 module right? If so, do we want to remove instead of deprecate?

Reply:
This is for Vault. Keeping it in but deprecated means that it's a part of the resulting struct, so you can unwrap v1-encoded structs with the v2 struct (I have a test for this in Boundary that encrypts in v1 and decrypts in v2 and vice versa, although it needs to be uncommented to run so as not to pull in both versions as deps). Eventually if Vault is no longer using this bool (instead using plaintext/ciphertext fields) then we can remove and reserve the number.

Reply:
Ah ok, this makes sense, thanks!